packages/replay/src/index.ts

```diff
@@ -1,18 +1,60 @@
-export { replayIntegration } from './integration';
-
-export type {
-  ReplayConfiguration,
-  ReplayEventType,
-  ReplayEventWithTime,
-  ReplayBreadcrumbFrame,
-  ReplayBreadcrumbFrameEvent,
-  ReplayOptionFrameEvent,
-  ReplayFrame,
-  ReplayFrameEvent,
-  ReplaySpanFrame,
-  ReplaySpanFrameEvent,
-  CanvasManagerInterface,
-  CanvasManagerOptions,
-} from './types';
-
-export { getReplay } from './util/getReplay';
+import {
+  getReplay as internalGetReplay,
+  replayIntegration as internalReplayIntegration,
+} from '@sentry-internal/replay';
+import type {
+  CanvasManagerInterface as InternalCanvasManagerInterface,
+  CanvasManagerOptions as InternalCanvasManagerOptions,
+  ReplayBreadcrumbFrame as InternalReplayBreadcrumbFrame,
+  ReplayBreadcrumbFrameEvent as InternalReplayBreadcrumbFrameEvent,
+  ReplayConfiguration as InternalReplayConfiguration,
+  ReplayEventType as InternalReplayEventType,
+  ReplayEventWithTime as InternalReplayEventWithTime,
+  ReplayFrame as InternalReplayFrame,
+  ReplayFrameEvent as InternalReplayFrameEvent,
+  ReplayOptionFrameEvent as InternalReplayOptionFrameEvent,
+  ReplaySpanFrame as InternalReplaySpanFrame,
+  ReplaySpanFrameEvent as InternalReplaySpanFrameEvent,
+} from '@sentry-internal/replay';
+
+/** @deprecated Import from `@sentry-internal/replay` */
+export const getReplay = internalGetReplay;
```

Comment:
m: Since this is (potentially) user-facing, the replacement should be to import from `@sentry/browser` or framework SDK instead of from `@sentry-internal/replay`.

For us internally, it's fine to eslint-ignore the deprecations in `v7` as long as we don't have the `@sentry-internal/replay` package.